objstore.go

```diff
@@ -189,6 +244,20 @@ type ObjectAttributes struct {
 	LastModified time.Time `json:"last_modified"`
 }
 
+type IterObjectAttributes struct {
+	Name         string
+	lastModified time.Time
+}
+
+func (i *IterObjectAttributes) SetLastModified(t time.Time) {
+	i.lastModified = t
+}
```

Comment:
Curious question: why did we make assignments to lastModified come through a setter but not name?

Reply:
i wanted to make it explicit that `lastModified` might not always be set, made it a private field so the getter contract can return `time.Time, bool`